tests/performance/test_mongo_redis_performance.py

```diff
@@ -0,0 +1,197 @@
+import random
+import time
+from random import randint
+from typing import Any, Callable, Dict
+
+import pytest
+
+from hyperon_das import DistributedAtomSpace
+from tests.integration.helpers import _db_down, _db_up, mongo_port, redis_port
+
+
+def measure(func):
+    def wrapper(*args, **kwargs):
+        start = time.time()
+        values = func(*args, **kwargs)
+        end = time.time()
+        TestPerformance.time = end - start
+        print(f'Elapsed time: {TestPerformance.time}')
+        return values
+
+    return wrapper
+
+
+class TestPerformance:
+    time = 0
+
+    @pytest.fixture(autouse=True)
+    def _initialize(
+        self,
+        node_range,
+        word_range,
+        letter_range,
+        alphabet_range,
+        word_link_percentage,
+        letter_link_percentage,
+        seed,
+    ):
+        self.node_type = 'Concept'
+        self.node_range = [int(i) for i in node_range.split('-')]
+        self.word_range = [int(i) for i in word_range.split('-')]
+        self.letter_range = [int(i) for i in letter_range.split('-')]
+        self.alphabet_range = [int(i) for i in alphabet_range.split('-')]
+        self.word_link_percentage = word_link_percentage
+        self.letter_link_percentage = letter_link_percentage
+        self.seed = seed
+        if seed:
+            random.seed(seed)
+
+    @pytest.fixture(scope="class", autouse=True)
+    def database(self):
+        _db_up()
+        yield
+        _db_down()
+
+    @staticmethod
+    def combinatorial_loop(d: Dict[str, Any], f: Callable[[int, int], None], percentage: float):
+        for v in d.values():
+            v = list(v)
+            for i, _ in enumerate(v):
+                for j in range(i + 1, len(v)):
+                    if random.random() > percentage:
+                        continue
+                    f(v[i], v[j])
+
+    @measure
+    def generate_links_word(self, word_dict):
+        links_dict_word = {}
+        word_count = self.word_range[1] - self.word_range[0]
+        for v in word_dict.values():
+            v = list(v)
+            for i in range(0, len(v)):
+                for j in range(i + 1, len(v)):
+                    if random.random() > self.word_link_percentage:
+                        continue
+                    key = f'{min(v[i], v[j])}->{max(v[i], v[j])}'
+                    if key in links_dict_word:
+                        links_dict_word[key] += 1 / word_count
```

Comment:
This sum is unstable. You should make all the adds first and then 1 division instead of summing a lot of divisions. The way you did, you potentialize rounding errors unnecessarily. 

You could just add +1 here and before returning the dict you can do an extra loop dividing every count by `word_count`.

Reply:
Moved division to function that adds the links to DAS